docs/dev/devel-env.rst

```diff
@@ -286,12 +294,147 @@ Also, run the **integration tests** for ListenBrainz.
+
+.. code-block:: bash
+
+    cp listenbrainz_spark/config.py.sample listenbrainz_spark/config.py
+
+
+Initialize ListenBrainz Spark containers
+----------------------------------------
+
+Next, run ``develop.sh spark build`` in the root of the repository. Using
+``docker-compose``, it creates multiple Docker containers for the different
+services and parts of the ListenBrainz Spark. This script starts Hadoop,
+Spark, and the ListenBrainz Playground containers. This also makes it easy to stop
+them all later.
+
+The first time you run it, it downloads and creates the containers. But it's not
+finished yet.
+
+.. code-block:: bash
+
+    ./develop.sh spark build
+    
+
+Format the NameNode container
+-----------------------------
+
+If it's your first time initializing the containers, format the ``namenode`` container
+using the given command.
+
+.. code-block:: bash
+
+    ./develop.sh format
+
+This will format the ``namenode`` and link the ``namenode`` and ``datanode`` to a single ClusterID. 
+
+**Warning:** Running this command more than once will erase all the ``namenode`` data and its 
+link with the ``datanode`` will be lost, as both of them will have different ClusterID's.
+
+Your development environment is now ready. Now, let's actually see ListenBrainz Spark
+in action!
+
+
+Run the magic script
+--------------------
+
+Start your development environment by executing ``develop.sh spark up``. Now, it 
+will work as expected.
+
+.. code-block:: bash
+
+    ./develop.sh spark up
+
+Now, you are all set to begin making changes and seeing them in real-time inside
+of your development environment!
+
+Once you are done with your work, shut down the containers using the following command. 
+
+.. code-block:: bash
+
+    ./develop.sh spark down
+
+
+Test your changes with unit tests
 ---------------------------------
 
-* What to do if getting an error while running './develop.sh build' command, 'ERROR: Couldn't connect to Docker daemon at http+docker://localhost - is it running?' ?
+Unit tests are an important part of ListenBrainz Spark. It helps make it easier for
+developers to test changes and help prevent easily avoidable mistakes later on.
+Before committing new code or making a pull request, run the unit tests on your
+code.
 
--  You need to add the user to the docker group by entering the following command in the terminal :
-  `sudo usermod -aG docker $USER`. After this command, restart the computer and then again run the './develop.sh build'.
+.. code-block:: bash
 
+   ./spark_test.sh
+
+This builds and runs the containers needed for the tests. This script configures
+test-specific data volumes so that test data is isolated from your development
+data.
+
+When the tests complete, you will see if your changes are valid or not. These tests
+are a helpful way to validate new changes without a lot of work.
+
+
+FAQs
```

Comment:
I think moving the faqs to a different file and linking it here is a good idea.

Reply:
Done